### PR TITLE
Artwr/add header handling

### PR DESCRIPTION
**Pull request by artwr**
@mistercrunch @krishnabhupatiraju 
This PR adds handling of headers for files imported from s3.
An additional parameter has been added that allows for the checking of the headers against the field names provided in the ordered dictionary that is passed to the operator. This allows for a very crude checking that the name + order of the columns in the file matches the one defined in the operator.

I attached in an email the test plan for this.

### CODE REVIEW

**Comment by mistercrunch** [comment]
Would `data/airflow/contrib/tests/` be a reasonable place for you test?

File: airflow/operators/s3_to_hive_operator.py [hunk 1]
```diff
@@ -84,17 +92,50 @@ def execute(self, context):
...
+                    self.hive_table,
+                    field_dict=self.field_dict,
+                    create=self.create,
+                    partition=self.partition,
+                    delimiter=self.delimiter,
+                    recreate=self.recreate)
+            else:
+                with open(f.name,'r') as tmpf:
+                    if self.check_headers:
+                        tmpf.seek(0)
```
Comment (mistercrunch) [inline airflow/operators/s3_to_hive_operator.py]: needed?

**Comment by artwr** [comment]
I hadn't thought about it, but we could put them there for now. I could also add unit tests for the S3 hooks.

File: airflow/operators/s3_to_hive_operator.py (hunk 1 above)
Comment (artwr) [inline airflow/operators/s3_to_hive_operator.py]: Ineed not. Removing it now.

**Comment by mistercrunch** [comment]
It would be nice to have move the header skipping into the load_file method of the HiveHook. It sucks to have to do a complete I/O roundtrip just to remove a header.

**Comment by artwr** [comment]
Agreed. With hive 0.13, it will just be adding a skip.line statement to the create table statement. This should definitely go into the hook.

But until then, I am not sure what to use other than this, or a sed '1d' which still requires IO round trip...

File: airflow/operators/s3_to_hive_operator.py [hunk 2]
```diff
@@ -84,17 +92,50 @@ def execute(self, context):
...
+                            raise Exception("Headers do not match the "
+                                            "field_dict keys")
+                    with NamedTemporaryFile("w") as f_no_headers:
+                        tmpf.seek(0)
+                        next(tmpf)
+                        for line in tmpf:
+                            f_no_headers.write(line)
+                        f_no_headers.flush()
+                        logging.info("Loading file without headers into Hive")
+                        self.hive.load_file(
```
Comment (mistercrunch) [inline airflow/operators/s3_to_hive_operator.py]: This is the same call as line 102 with a different filename, I'd take it out of the if statetment.

Comment (artwr) [inline airflow/operators/s3_to_hive_operator.py]: It's a different tempfile with a different with scope. Other wise I need to bring the scope up and always create 2 different tempfile, whether there is a header or not.

Comment (mistercrunch) [inline airflow/operators/s3_to_hive_operator.py]: oh right, the file disappears if you get out the out of the with block

**Pull request merged**